examples/tensorflow-profiling-examples/trainer-deep-profiling/model.py

```diff
@@ -0,0 +1,151 @@
+
+TRANSFORMED_TRAIN_DATA_FILEBASE = 'train_transformed'
+TRANSFORMED_TEST_DATA_FILEBASE = 'test_transformed'
+EXPORTED_MODEL_DIR = 'exported_model_dir'
+
+
+def read_dataset(mode):
+    def _decode_csv(line):
+        features = tf.decode_csv(line, record_defaults=DEFAULTS)
+        features = dict(zip(CSV_COLUMNS, features))
+        labels = features.pop(LABEL_COLUMN)
+        return features, labels
+
+    def _input_fn():
+        # use prefix to create filename
+        if mode == tf.estimator.ModeKeys.TRAIN:
+            prefix = 'train'
+            num_epochs = None
+            shuffle = True
+        else:
+            prefix = 'eval'
+            num_epochs = 1
+            shuffle = False
+        filename_pattern = 'gs://{}/preproc/{}*{}*'.format(BUCKET, prefix, PATTERN)
+        files = tf.data.Dataset.list_files(file_pattern=filename_pattern,
+                                           shuffle=shuffle)
+        dataset = files.flat_map(lambda filename: tf.data.TextLineDataset(filename).map(_decode_csv))
+        if mode == tf.estimator.ModeKeys.TRAIN:
+            dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)
+        else:
+            dataset = dataset.batch(5012)
+        return dataset.repeat(num_epochs).make_one_shot_iterator().get_next()
+
+    return _input_fn
+
+
+def get_wide_deep(n_embeds):
+    # define column types
+    plurality_values = ['Single(1)', 'Twins(2)', 'Triplets(3)',
+                        'Quadruplets(4)', 'Quintuplets(5)', 'Multiple(2+)']
+    is_male = tf.feature_column.categorical_column_with_vocabulary_list('is_male', ['True', 'False', 'Unknown'])
+    mother_age = tf.feature_column.numeric_column('mother_age')
+    plurality = tf.feature_column.categorical_column_with_vocabulary_list('plurality', plurality_values)
+    gestation_weeks = tf.feature_column.numeric_column('gestation_weeks')
+
+    # discretize
+    age_buckets = tf.feature_column.bucketized_column(mother_age,
+                                                      boundaries=np.arange(15, 45, 1).tolist())  # pylint: disable=C0301
+    gestation_buckets = tf.feature_column.bucketized_column(gestation_weeks,
+                                                            boundaries=np.arange(17, 47, 1).tolist())  # pylint: disable=C0301
+
+    # sparse columns are wide
+    wide = [is_male, plurality, age_buckets, gestation_buckets]
+
+    # feature cross all the wide columns and embed into a lower dimension
+    crossed = tf.feature_column.crossed_column(wide, hash_bucket_size=20000)
+    embed = tf.feature_column.embedding_column(crossed, n_embeds)
+
+    # continuous columns are deep
+    deep = [mother_age, gestation_weeks, embed]
+    return wide, deep
+
+
+def my_rmse(labels, predictions):
+    pred_values = predictions['predictions']
+    return {'rmse': tf.metrics.root_mean_squared_error(labels, pred_values)}
+
+
+def serving_input_fn():
+    feature_placeholders = {'is_male': tf.placeholder(tf.string, [None]),
```

Comment:
Please wrap to avoid going over the 80 column limit:

    feature_placeholders = {
        'is_male': tf.placeholder(tf.string, [None]),
        'mother_age': tf.placeholder(tf.float32, [None]),
        ...
    }

Reply:
ack